src/css.js

```diff
@@ -274,11 +286,16 @@ jQuery.extend( {
 
 	css: function( elem, name, extra, styles ) {
 		var val, num, hooks,
-			origName = jQuery.camelCase( name );
-
-		// Make sure that we're working with the right name
-		name = jQuery.cssProps[ origName ] ||
-			( jQuery.cssProps[ origName ] = vendorPropName( origName ) || origName );
+			origName = jQuery.camelCase( name ),
+			customProperty = isCustomProperty( name );
+
+		// Make sure that we're working with the right name. We don't
+		// want to modify the value if it is a CSS custom property
+		// since they are user-defined.
+		if ( !customProperty ) {
+			name = jQuery.cssProps[ origName ] ||
```

Comment:
Maybe you could simplify this part? We are trying to get rid of the past magic code